### PR TITLE
Prepare core releases for March 2023

### PR DESCRIPTION
**Pull request by samvaity**
# Description

Prepare Core libraries for their March 2023 releases.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/versioning/version_client.txt [hunk 1]
```diff
@@ -408,8 +408,8 @@ com.azure.tools:azure-sdk-build-tool;1.0.0-beta.1;1.0.0-beta.2
 # In the pom, the version update tag after the version should name the unreleased package and the dependency version:
 # <!-- {x-version-update;unreleased_com.azure:azure-core;dependency} -->
 
-unreleased_com.azure:azure-core-amqp;2.9.0-beta.1
-unreleased_com.azure:azure-core;1.37.0-beta.1
+unreleased_com.azure:azure-core-amqp;2.8.3
+unreleased_com.azure:azure-core;1.37.0
```
Comment (vcolin7) [inline eng/versioning/version_client.txt:412]: After these versions are released we should make sure to remove these entries from this file and update the POMs where they were used to use the `dependency` version instead.

Comment (vcolin7) [reply eng/versioning/version_client.txt:412]: That said, I'm not sure if the release pipeline is going to succeed for a dependency marked as unreleased. We might want to replace the tags in POMs with `current` instead.

File: sdk/core/azure-core-serializer-json-jackson/CHANGELOG.md [hunk 2]
```diff
@@ -9,11 +9,12 @@
...
 
-### Breaking Changes
+### Other Changes
 
-### Bugs Fixed
+#### Dependency Updates
 
-### Other Changes
+- Upgraded `azure-core` from `1.36.0` to `1.37.0`.
+- Upgraded Jackson to the latest releases `2.1.X`.
```
Comment (vcolin7) [inline sdk/core/azure-core-serializer-json-jackson/CHANGELOG.md:17]: We should specify which versions the Jackson dependencies were updated to, the latest one I can find in the CHANGELOG says `2.13.4`, so `2.1.x` seems misleading

File: sdk/core/azure-core/CHANGELOG.md [hunk 3]
```diff
@@ -22,7 +22,10 @@
 - Fixed a bug where HTTP headers and query parameters weren't included in HTTP request and response logging when
   `INFORMATIONAL` log level was used.
 
-### Other Changes
+### Dependency Updates
+
+- Upgraded Reactor Core from `3.4.26` to `3.4.27`.
+- Upgraded Jackson to the latest releases `2.1.X`.
```
Comment (vcolin7) [inline sdk/core/azure-core/CHANGELOG.md:28]: Same comment about Jackson versions.

Comment (srnagar) [inline sdk/core/azure-core/CHANGELOG.md:28]: Should this be `2.13.5`?

**Review by srnagar — APPROVED** [review]

File: eng/versioning/version_client.txt (hunk 1 above)
Comment (samvaity) [reply eng/versioning/version_client.txt:412]: You are right, release pipeline will not succeed if we have these around and should be removed in the Increment PR.

**Review by vcolin7 — APPROVED** [review]

**Pull request merged**